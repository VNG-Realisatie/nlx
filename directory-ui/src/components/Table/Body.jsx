import styled from 'styled-components'

const Body = styled.tbody`
    display: table-row-group;
`

export default Body
