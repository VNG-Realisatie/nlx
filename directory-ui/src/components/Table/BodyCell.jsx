import { node, oneOf } from 'prop-types'
import styled from 'styled-components'

const BodyCell = styled.td`
    display: table-cell;
    padding: .6rem;

    font-size: 1rem;
    line-height: 1rem;
    font-weight: 400;
    
    text-align: ${p => p.align};
`

BodyCell.propTypes = {
  children: node,
  align: oneOf(['left', 'center', 'right'])
}

BodyCell.defaultProps = {
  align: 'left',
}

export default BodyCell
