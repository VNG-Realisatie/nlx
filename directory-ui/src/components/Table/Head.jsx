// Copyright Â© VNG Realisatie 2020
// Licensed under the EUPL
//

import { node } from 'prop-types'
import styled from 'styled-components'

const Head = styled.thead`
  display: table-header-group;
`

Head.propTypes = {
  children: node,
}

export default Head
