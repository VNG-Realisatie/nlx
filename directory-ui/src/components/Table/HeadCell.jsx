import { node, oneOf } from 'prop-types'
import styled from 'styled-components'

const HeadCell = styled.th`
    display: table-cell;
    padding: .65rem;

    font-size: .85rem;
    line-height: 1.25rem;
    font-weight: 600;
    color: #ADB3C6;
    overflow: hidden;
    text-overflow: ellipsis;
    text-transform: uppercase;
    white-space: nowrap;
    
    text-align: ${ p => p.align};
`

HeadCell.propTypes = {
  children: node,
  align: oneOf(['left', 'center', 'right'])
}

HeadCell.defaultProps = {
  align: 'left'
}

export default HeadCell
