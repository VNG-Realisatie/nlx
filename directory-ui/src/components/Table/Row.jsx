// Copyright Â© VNG Realisatie 2018
// Licensed under the EUPL

import styled from 'styled-components'

const Row = styled.tr`
    display: table-row;

    &:first-child {
      td {
        &:first-child {
          border-top-left-radius: 3px;
        }

        &:last-child {
          border-top-right-radius: 3px;
        }
      }
    }

    &:not(:last-child) {
      td {
        border-bottom: 1px solid #F0F2F7;
      }
    }

    &:last-child {
      td {
        &:first-child {
          border-bottom-left-radius: 3px;
        }

        &:last-child {
          border-bottom-right-radius: 3px;
        }
      }
    }
`

export default Row
