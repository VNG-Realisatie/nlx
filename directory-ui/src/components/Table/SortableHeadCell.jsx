import React from 'react'
import { oneOf } from 'prop-types'
import { StyledSortableTableHeadCell, StyledArrow } from './SortableHeadCell.styles'

export const ASCENDING = 'asc'
export const DESCENDING = 'desc'

const ArrowUp = () =>
  <StyledArrow viewBox="0 0 8 10">
    <path d="M5 6V0H3v6H0l4 4 4-4z" fill="#2D3240" />
  </StyledArrow>

const ArrowDown = () =>
  <StyledArrow viewBox="0 0 8 10">
    <path d="M3 4v6h2V4h3L4 0 0 4z" fill="#2D3240" />
  </StyledArrow>

const SortableArrow = ({ direction }) =>
  direction === ASCENDING ? <ArrowDown/> : <ArrowUp/>

const SortableHeadCell = ({ children, direction, ...other }) =>
  <StyledSortableTableHeadCell isSorting={direction} {...other}>
    {children}
    {direction ? <SortableArrow direction={direction} /> : null}
  </StyledSortableTableHeadCell>

SortableHeadCell.propTypes = {
  direction: oneOf([ASCENDING, DESCENDING, null])
}

export default SortableHeadCell
