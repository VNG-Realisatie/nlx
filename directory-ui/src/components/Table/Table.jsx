import { node } from 'prop-types'
import styled from 'styled-components'

const Table = styled.table`
    border-collapse: collapse;
    border-spacing: 0;
    display: table;
    width: 100%;
`

Table.propTypes = {
  children: node,
}

export default Table
