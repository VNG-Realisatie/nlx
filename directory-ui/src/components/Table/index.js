import { node } from 'prop-types'
import styled from 'styled-components'

import Body from './Body'
import HeadCell from './HeadCell'
import BodyCell from './BodyCell'
import Head from './Head'
import Row from './Row'
import SortableHeadCell from './SortableHeadCell'

const Table = styled.table`
    border-collapse: separate;
    border-spacing: 0;
    display: table;
    margin: 0 auto;
`

Table.propTypes = {
  children: node,
}

Table.Body = Body
Table.HeadCell = HeadCell
Table.BodyCell = BodyCell
Table.Head = Head
Table.Row = Row
Table.SortableHeadCell = SortableHeadCell

export default Table
