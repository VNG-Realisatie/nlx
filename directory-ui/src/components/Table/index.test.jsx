// Copyright Â© VNG Realisatie 2018
// Licensed under the EUPL

import React from 'react'
import { shallow } from 'enzyme'
import Table from './index'

it('should render child elements', () => {
  expect(shallow(<Table>
    <tr>
      <td>Table body</td>
    </tr>
  </Table>).contains(<tr>
    <td>Table body</td>
  </tr>)).toEqual(true)
})
