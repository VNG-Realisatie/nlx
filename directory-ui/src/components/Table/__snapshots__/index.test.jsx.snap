// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "Body": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    display: table-row-group;
",
          ],
        },
        "displayName": "styled.tbody",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "tbody",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "BodyCell": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "rules": Array [
            "
    display: table-cell;
    padding: .6rem;

    font-size: 1rem;
    line-height: 1rem;
    font-weight: 400;
    
    text-align: ",
            [Function],
            ";
",
          ],
        },
        "defaultProps": Object {
          "align": "left",
        },
        "displayName": "styled.td",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "align": [Function],
          "children": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "td",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Head": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
    display: table-header-group;
",
          ],
        },
        "displayName": "styled.thead",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "children": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "thead",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "HeadCell": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
    display: table-cell;
    padding: .65rem;

    font-size: .85rem;
    line-height: 1.25rem;
    font-weight: 600;
    color: #ADB3C6;
    overflow: hidden;
    text-overflow: ellipsis;
    text-transform: uppercase;
    white-space: nowrap;
    
    text-align: ",
            [Function],
            ";
",
          ],
        },
        "defaultProps": Object {
          "align": "left",
        },
        "displayName": "styled.th",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "align": [Function],
          "children": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "th",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Row": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": true,
          "rules": Array [
            "
    display: table-row;
    
    &:first-child {
      td {
        border-top: 1px solid #EDEFF1;
      }
    
      td {
        &:first-child {
          border-top-left-radius: 3px;
          border-left: 1px solid #EDEFF1;
        }
        
        &:last-child {
          border-top-right-radius: 3px;
          border-right: 1px solid #EDEFF1;
        }
      }
    }
    
    &:not(:last-child) {
      td {
        border-bottom: 1px solid #EDEFF1;
      }
    }
    
    &:not(:first-child):not(:last-child) {
      td {
        &:first-child {
          border-left: 1px solid #EDEFF1;
        }
        
        &:last-child {
          border-right: 1px solid #EDEFF1;
        }
      }        
    }
    
    &:last-child {
      td {
        border-bottom: 1px solid #EDEFF1;
      
        &:first-child {
          border-bottom-left-radius: 3px;
          border-left: 1px solid #EDEFF1;
        }
        
        &:last-child {
          border-bottom-right-radius: 3px;
          border-right: 1px solid #EDEFF1;
        }
      }
    }
",
          ],
        },
        "displayName": "styled.tr",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "tr",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "SortableHeadCell": [Function],
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": true,
        "rules": Array [
          "
    border-collapse: separate;
    border-spacing: 0;
    display: table;
    width: 100%;
",
        ],
      },
      "displayName": "styled.table",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "children": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "table",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
